### DISCUSSION TITLE
Taking 10 seconds to receive one message.

### DISCUSSION THREAD

**Discussion by Donadominic** [nanomq/nanomq#994] [Q&A] [answered]
I am publishing messages to mqtt topic while the subscriber is offline. But when the subscriber is active it is taking 10 seconds to receive each message. How to make it fast? I want to receive all the messages published while the subscriber is offline in minimum time.

**Comment by JaylinYu** [comment]
modify the "qos_duration" in nanomq.conf, like 1 sec if you want it to be really fast

**Reply by Donadominic** [reply]
docker run -d -p 1887:1883 -p 8883:8883 \
-e NANOMQ_QOS_DURATION=1 \
--name nanomq emqx/nanomq:latest

I set it with command. But still, it is taking 10 seconds.

**Reply by JaylinYu** [reply]
How many messages take 10 seconds?  If the config takes effect, you should be able to see the client receiving cached msg at 1/s rate.

**Reply by Donadominic** [reply]
one message in 10 seconds

**Reply by JaylinYu** [reply]
Apparently, the env variable is not valid. I will look into it, while you can try with config file.

**Reply by JaylinYu** [reply]
I just merged a PR to fix this. I can release a new version for docker If you want

**Comment by JaylinYu** [comment, marked as answer]
@Donadominic Seems like config file will overwrite environmental variables, plz use config file for now.

**Reply by JaylinYu** [reply]
This has been fixed in 0.15.1 docker image. Now env variables also takes effect.
